### DISCUSSION TITLE
Spellchecking in SILE with vim

### DISCUSSION THREAD

**Discussion by ctrlcctrlv** [sile-typesetter/sile#1557] [Show and tell]
```vim
" Spellchecking
Plug 'melvio/medical-spell-files' " Assumes plug.vim
function FRBSetUpSpell()
    setlocal spell
    call plug#load('medical-spell-files') " Assumes plug.vim
    setlocal spelllang=en_us,cjk,medical
    setlocal spelloptions=camel
    " Generated with ↓
    " s/\v%(spellcapcheck\=)\zs(.*)/\=substitute(substitute(submatch(1), '\\', '\\\\', 'g'), '[ "'."'".'|]', '\\\0', 'g')
    " Against        ↓
    " \v([a-z]\.[a-z]|№|\.)@<![.?!]\_[\])'" \t]+
    setlocal spellcapcheck=\\v([a-z]\\.[a-z]\|№\|\\.)@<![.?!]\\_[\\])\'\"\ \\t]+
endfunction
au BufRead *.{sil,md,txt,markdown} call FRBSetUpSpell() | syntax spell toplevel
```

![image](https://user-images.githubusercontent.com/838783/188281498-19f320e2-d1e9-4325-a7de-37b06a0f08fe.png)
![image](https://user-images.githubusercontent.com/838783/188281510-9e7bc9e0-125e-4e7e-983a-d90bf147375d.png)

**Comment by alerque** [comment]
Ouch, this reminds me I fixed spell checking for content regions in SIL files in my vim locally but didn't post it to vim-sile. Stand by.

**Comment by alerque** [comment]
I just cleaned up and committed a bunch of changed I had in my local copy of `vim-sile`. It now handles SILE v0.14+ stuff like `\lua`, `\xml`, and `\raw`. Additionally it fixes the spell checking so that comments and content are spell checked but not commands or options.
